### ISSUE TITLE
Author and editor roles

### ISSUE THREAD

**Issue by dlschwartz** [srophe/srophe-app-data#740]
@nathangibson I'm working on the Syriaca.org ODD at the moment as we try to work up some better documentation. I came across this line of the current ODD (https://github.com/srophe/srophe-eXist-app/blob/master/documentation/syriaca-tei-main.odd#L569) but the URL doesn't work. Has this been moved? Is there some other list? 

Do you have any thoughts regarding the question (you?) posed here: https://github.com/srophe/srophe-eXist-app/blob/master/documentation/syriaca-tei-main.odd#L565? I think it would be great to firm things up a bit in the schema. I'm happy to get it working and documented in the ODD.

Oh, and please let me know if you're not the right person for this question. Thanks.

**Comment by nathangibson** [comment]
1. The file exists here: https://github.com/srophe/srophe-eXist-app/blob/master/documentation/author-editor-roles.xml and here: https://github.com/srophe/srophe-eXist-app/blob/master/documentation/author-editor-roles.xml. @wsalesky maybe it is somehow corrupted in eXist?
2. I wonder if we should maybe make the ODD file the place for documenting these roles and get rid of the other XML file. It would make the maintenance simpler. But I wonder if that is sufficiently transparent to the end user? Would they automatically know to look in the ODD file for the documentation? In fact, I don't see that we link to our ODD file in our individual TEI records. Should we be doing that?

**Comment by dlschwartz** [comment]
@nathangibson Thanks for the link. 

The goal we are working toward is to have a prose documentation (draft of places here: https://github.com/srophe/srophe-app-data/wiki/TEI-Encoding-Manual-for-The-Syriac-Gazetteer) where the element links link out to html output from the ODD that will produce customized pages for each element like you are used to seeing in the TEI guidelines. These will include customized Syriaca.org descriptions of the elements and examples from the project. This would mean that there will be two ways to see the documentation but beyond that the customized schema will require many of the things that our current documentation states as required even if the schema doesn't require it for validation in oxygen.

@wsalesky and I have exchanged an email about doing some indexing of Syriaca.org data that would allow us to point to a single xml file (rather than rifling through 1000s of separate xml files) for purposes of validation. You can see an example of what I have in mind with SPEAR and the ODD pointing to an index of the taxonomy: https://github.com/srophe/srophe-app-data/blob/dev/data/subjects/taxonomyIndex.xml. Now that I have that index, I can use schematron to point to pieces of that index with my ODD in specific contexts. For example, lines 1844-1865 of this link (https://github.com/srophe/srophe-eXist-app/blob/schema-4-spear-edits/srophe-app/documentation/SPEARschema/odd_SPEAR.xml#L1844) are how I allow only occupation keywords as the value of @ref attributes when they appear on <occupation> elements.

Implementing something similar more broadly in the ODD would, for example, create an error (red) or warning (yellow) message to pop up in oxygen if an encoder input a place URI that wasn't among the place URIs on dev. It would require that an author @ref pointed to a person URI that is encoded as an author. This would prevent typos, help with consistent data encoding across modules, and function as a nice aid to editing new records.

**Comment by dlschwartz** [comment]
@nathangibson Thanks for the links! We're working on these kinds of documentation issues right now. Specifically, we're working toward a documentation that will look a lot like the TEI guideline but specific to Syriaca.org. Here you can see a draft of some developments to a prose documentation for place data: https://github.com/srophe/srophe-app-data/wiki/TEI-Encoding-Manual-for-The-Syriac-Gazetteer. I'm working to bring the ODD into line with the prose descriptions, to require in the ODD/schema a number of the things that we just state are required/preferred in descriptions, and to get all of the "context" stuff working for validation in oxygen. Once this is done we will output an html version from the ODD that will produce something that looks very similar to the element pages in the TEI guidelines but with rows customized for Syriaca.org: attributes, contained by, may contain, notes, examples, etc. The prose documentation will link to the element documentation just as it does in the TEI guidelines. For another related development, see a new issue I just opened up: https://github.com/srophe/srophe-app-data/issues/741.

**Comment by nathangibson** [comment]
I like it!
